tpu/hptuning/resnet-hptuning/resnet/imagenet_input.py

```diff
@@ -0,0 +1,211 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Efficient ImageNet input pipeline using tf.data.Dataset."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+
+import tensorflow as tf
+
+from . import resnet_preprocessing
+
+
+def image_serving_input_fn():
```

Comment:
Why is this function not returning anything. Also why is there a need to have ServingInputReceiver in an inner function and not directly in serving?

Reply:
The hptuning example is entirely based on the official ResNet TPU sample version r1.8 and I made only minimal changes required to use CMLE's hptuning service.  I would prefer to minimize diverging from the official (training) sample.